### PR TITLE
chore: add automaxprocs to limit parallel processes inside containers

### PR DESCRIPTION
**Pull request by kylecarbs**
I saw this post on HackerNews: https://www.riverphillips.dev/blog/go-cfs/. It seems like we could benefit from this as well, because Coder is so frequently deployed in a container.

### CODE REVIEW

**Review by mafredri — APPROVED** [review]
Nice! I had been meaning to add this, but I’m glad you did because I forgot. Should we do the agent as well?

**Comment by kylecarbs** [comment]
Great idea. Will do!

**Comment by mafredri** [comment]
@kylecarbs I'm re-opening this since I'm good with having this merged. Seems there's a tiny lint issue remaining, but raising the bar cyclo limit seems fine for this PR (+1).

File: cli/agent.go [hunk 1]
```diff
@@ -162,6 +162,12 @@ func (r *RootCmd) workspaceAgent() *clibase.Cmd {
 			sinks = append(sinks, sloghuman.Sink(logWriter))
 			logger := slog.Make(sinks...).Leveled(slog.LevelDebug)
 
+			undoMacProcs, err := limitGoMaxProcs(logger)
+			if err != nil {
+				return xerrors.Errorf("set maxprocs: %w", err)
+			}
+			defer undoMacProcs()
```
Comment (mafredri) [inline cli/agent.go:169]:
```suggestion
			undoMaxProcs, err := limitGoMaxProcs(logger)
			if err != nil {
				return xerrors.Errorf("set maxprocs: %w", err)
			}
			defer undoMaxProcs()
```

**Pull request closed without merging**